src/Whim/Workspace/WorkspaceManager.cs

```diff
@@ -402,12 +402,35 @@ public void ActivatePrevious(IMonitor? monitor = null)
 		int idx = _workspaces.IndexOf(currentWorkspace);
 		int prevIdx = (idx - 1).Mod(_workspaces.Count);
 
+		if (skipActive)
+		{
+			if (_workspaces.Count <= _context.MonitorManager.Length)
+			{
+				Logger.Debug($"No inactive workspace found");
+				return;
+			}
+			foreach (IMonitor m in _context.MonitorManager)
+			{
+				IWorkspace? w = GetWorkspaceForMonitor(m);
+				if (w == null)
+				{
+					Logger.Debug($"No workspace found for monitor {m}");
+					return;
+				}
+				int wIdx = _workspaces.IndexOf(w);
```

Comment:
I think it'd be possible to do it in a slightly different way, such that each iteration doesn't have this `O(n)` loop

Reply:
`active.Contains(idx)` is still `O(n)`. 

I'd iterate over the workspaces, then check if the current workspace has an assigned workspace using `GetWorkspaceForMonitor`. If it doesn't, I'd activate that one. If we get back to `currentIdx == startIdx`, then we wouldn't do anything because the number of workspaces == number of monitors.